Sources/Swift/Tools/SentryLog.swift

```diff
@@ -0,0 +1,75 @@
+@_implementationOnly import _SentryPrivate
+import Foundation
+
+@objc
+class SentryLog: NSObject {
+    
+    static private(set) var isDebug = true
+    static private(set) var diagnosticLevel = SentryLevel.error
+    private static var logOutput = SentryLogOutput()
+    private static var logConfigureLock = NSLock()
+
+    @objc
+    static func configure(_ isDebug: Bool, diagnosticLevel: SentryLevel) {
+        logConfigureLock.synchronized {
+            self.isDebug = isDebug
+            self.diagnosticLevel = diagnosticLevel
+        }
+        sentry_initializeAsyncLogFile()
+    }
+    
+    @objc
+    static func log(message: String, andLevel level: SentryLevel) {
+        guard willLog(atLevel: level) else { return }
+        logOutput.log("[Sentry] [\(level)] \(message)")
+    }
+
+    /**
+     * @return @c YES if the current logging configuration will log statements at the current level,
+     * @c NO if not.
+     */
+    @objc
+    static func willLog(atLevel level: SentryLevel) -> Bool {
+        return isDebug && level != .none && level.rawValue >= diagnosticLevel.rawValue
+    }
+ 
+    #if TEST || TESTCI
+    
+    static func setOutput(_ output: SentryLogOutput) {
+        logOutput = output
+    }
+    
+    static func getOutput() -> SentryLogOutput {
+        return logOutput
+    }
+    
+    #endif
+}
+
+extension SentryLog {
+    private static func log(level: SentryLevel, message: String, file: String, line: Int) {
+        let path = file as NSString
```

Comment:
`h`: We should also guard this with an early exit to avoid the extra work of allocating a string. I added this with https://github.com/getsentry/sentry-cocoa/pull/4028 and allocating extra strings in log messages caused problems previously, see https://github.com/getsentry/sentry-cocoa/pull/4028.
```suggestion
        guard willLog(atLevel: level) else { return }
        let path = file as NSString
```